selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/stack/CreateWorkspaceFromJavaCentosStackTest.java

```diff
@@ -0,0 +1,140 @@
+  @Inject private Dashboard dashboard;
+  @Inject private CheTerminal terminal;
+  @Inject private DefaultTestUser defaultTestUser;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private CreateWorkspaceHelper createWorkspaceHelper;
+  @Inject private TestWorkspaceServiceClient workspaceServiceClient;
+
+  @BeforeClass
+  public void setUp() {
+    dashboard.open();
+  }
+
+  @AfterClass
+  public void tearDown() throws Exception {
+    workspaceServiceClient.delete(WORKSPACE_NAME, defaultTestUser.getName());
+  }
+
+  @Test
+  public void checkWorkspaceCreationFromJavaCentosStack() {
+    createWorkspaceHelper.createWorkspaceFromStackWithProjects(
+        JAVA_CENTOS, WORKSPACE_NAME, projects);
+
+    ide.switchToIdeAndWaitWorkspaceIsReadyToUse();
+
+    projectExplorer.waitProjectInitialization(CONSOLE_JAVA_SIMPLE);
+    projectExplorer.waitProjectInitialization(WEB_JAVA_SPRING);
+  }
+
+  @Test(priority = 1)
+  public void checkConsoleJavaSimpleProjectCommands() {
+    consoles.executeCommandFromProjectExplorer(
+        CONSOLE_JAVA_SIMPLE, BUILD_GOAL, BUILD_COMMAND, BUILD_SUCCESS);
+
+    consoles.executeCommandFromProjectExplorer(
+        CONSOLE_JAVA_SIMPLE,
+        BUILD_GOAL,
+        BUILD_COMMAND_ITEM.getItem(CONSOLE_JAVA_SIMPLE),
+        BUILD_SUCCESS);
+
+    consoles.executeCommandFromProjectExplorer(
+        CONSOLE_JAVA_SIMPLE,
+        RUN_GOAL,
+        RUN_COMMAND_ITEM.getItem(CONSOLE_JAVA_SIMPLE),
+        "Hello World Che!");
+  }
+
+  @Test(priority = 1)
+  public void checkWebJavaSpringProjectCommands() {
+    consoles.executeCommandFromProjectExplorer(
+        WEB_JAVA_SPRING, BUILD_GOAL, BUILD_COMMAND, BUILD_SUCCESS);
+
+    consoles.executeCommandFromProjectExplorer(
+        WEB_JAVA_SPRING, BUILD_GOAL, BUILD_COMMAND_ITEM.getItem(WEB_JAVA_SPRING), BUILD_SUCCESS);
+
+    consoles.executeCommandFromProjectExplorer(
+        WEB_JAVA_SPRING,
+        RUN_GOAL,
+        BUILD_AND_RUN_COMMAND_ITEM.getItem(WEB_JAVA_SPRING),
+        SERVER_STARTUP_IN);
+    consoles.checkWebElementVisibilityAtPreviewPage(By.xpath("//span[text()='Enter your name: ']"));
+    consoles.closeProcessTabWithAskDialog(BUILD_AND_RUN_COMMAND_ITEM.getItem(WEB_JAVA_SPRING));
+
+    consoles.executeCommandFromProjectExplorer(
+        WEB_JAVA_SPRING,
+        RUN_GOAL,
+        RUN_TOMCAT_COMMAND_ITEM.getItem(WEB_JAVA_SPRING),
+        SERVER_STARTUP_IN);
+    consoles.checkWebElementVisibilityAtPreviewPage(By.xpath("//span[text()='Enter your name: ']"));
+
+    // execute 'stop tomcat' command and check that tomcat is not running
+    projectExplorer.invokeCommandWithContextMenu(
+        RUN_GOAL, WEB_JAVA_SPRING, STOP_TOMCAT_COMMAND_ITEM.getItem(WEB_JAVA_SPRING));
+    consoles.selectProcessInProcessConsoleTreeByName("Terminal");
+    terminal.typeIntoTerminal("ps ax");
+    terminal.typeIntoTerminal(ENTER.toString());
+    terminal.waitExpectedTextNotPresentTerminal("/bin/bash -c $TOMCAT_HOME/bin/catalina.sh");
```

Comment:
It's not quite clear what `"/bin/bash -c $TOMCAT_HOME/bin/catalina.sh"` is about. From this point it would be better to name the string.